src/main.rs

```diff
@@ -337,16 +344,24 @@ fn todo(req: HttpRequest<AppState>) -> actix_web::HttpResponse {
     use std::borrow::Borrow;
     let renderer: &htmlrenderer::HTMLRenderer = req.state().html_renderer.borrow();
     let username = req.identity().unwrap_or(String::new());
+    let current_theme = match req.cookie("theme") {
+        Some(x) => String::from(x.value()),
+        None => String::new(),
+    };
```

Comment:
Yeah, I'll see if I can wrap my head around the syntax and such. I have never done any Rust!

But, I'm looking at the code now and... isn't `logged_in` always `true`?

Reply:
Ah, I see now that `logged_in` is not necessarily always `true`, because of the `username.len()` in the `markdown_page` calls.

Anyway, I'm looking more into it and I am a bit unsure, but I'll do my best.

Reply:
Is it that useful to have the themes in that `enum`? It seems that they make everything messier, and essentially we have to convert back and forth all the time, eg. you cannot pass an enum value directly to `json!`, so we would have to convert `SessionData` to a regular hashmap before it could be merged with the rest and passed to `json!`...

What do you think? I don't even know if I have the Rust chops to pull this off :flushed: